wordpress-to-github/index.js

```diff
@@ -473,9 +494,51 @@ const SyncEndpoint = async (
     );
   }
 
+  // API Requests
+  if (allApiRequests) {
+    // Group all destination files by their parent folders.
+    const apiRequestsByFolder = allApiRequests.reduce((bucket, request) => {
+      let folderName = request.Destination.split("/").slice(0, -1).join("/");
+      let fileName = request.Destination.split("/").slice(-1);
+
+      if (!(folderName in bucket)) {
+        bucket[folderName] = new Map();
+      }
+
+      bucket[folderName].set(fileName, JSON.stringify(request.Data, null, 2))
+
+      return bucket;
+    }, {});
+
+    // Create and commit a git tree for each set of files.
+    for (let [folderName, fileMap] of Object.entries(apiRequestsByFolder)) {
+      const requestsTree = await createTreeFromFileMap(
```

Comment:
In my tests It creates a tree every time.  This might be a problem with the tree code update.  Creating a new tree when nothing changes could be problematic. "Creating tree for Wordpress API Requests Update (1 update to xx) - 1/1 items"

Reply:
A possible solution could involve the cache-check step. We could remove a file from `allApiRequests` when it's matched in the cache. That way, we wouldn't create a tree later for files that don't need to be updated. 

It's probably not a quick fix though. And it'd only work when there's a cache to check, so not on App Function cold starts.

Reply:
Fixed.  The filename that was being added to the Map was an array instead of a string.